### PR TITLE
call modules recursively

### PR DESCRIPTION
**Pull request by justinemarie**
This switches back to calling modules recursively to improve throughput.

### CODE REVIEW

File: core/module.h [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include "task.h"
 
 static inline void set_cmd_response_error(pb_cmd_response_t *response,
-                                          const pb_error_t &error) {
+    const pb_error_t &error) {
```
Comment (changlan) [inline core/module.h]: Nit: Unnecessary indentation changes to the file (along with other lines below)

Comment (justinemarie) [inline core/module.h]: I ran clang-format -- is that no longer recommended?

Comment (changlan) [inline core/module.h]: That doesn't seem like the clang-format output. Perhaps this file is missed? (just guessing)

**Comment by justinemarie** [comment]
Looks like that quirk is gone now that I actually clang-formatted

**Comment by barath** [comment]
Dumb question -- where are the stack-allocated packet batches in this PR?

**Comment by justinemarie** [comment]
They are allocated in RunTask or in ProcessBatch (in the modules themselves).

**Comment by barath** [comment]
Are no changes needed in RunTask or ProcessBatch then to revert to that behavior?

**Comment by justinemarie** [comment]
That's correct.
* The current code in the C++ branch allows them to pass along a pointer to their stack-allocated PacketBatch to RunChooseModule, and then RunChooseModule copies it in to the heap and returns. So they were always stack-allocating and then we just copied them.
* The new-new version that was never committed to the C++ allocated everything from the mempool in the heap, and that required changes to all the modules. But I never made a pull request for that version since it still wasn't as fast as the recursive version.

**Comment by justinemarie** [comment]
What is this failure?

**Comment by justinemarie** [comment]
image

**Comment by melvinw** [comment]
cat
Thanks!

**Pull request merged**